### ISSUE TITLE
Platforms retry with no hosts available

### ISSUE THREAD

**Issue by hjoliver** [cylc/cylc-flow#5442]
### Description

Platforms config:
```ini
# global.cylc
[platforms]
   [[nohosts-1]]  # (platform name does not match a host name)
       install target = localhost
   [[nohosts-2]]  # (platform name does not match a host name)
       install target = localhost
```

Workflow config:
```ini
[scheduling]
   [[graph]]
      R1 = "foo & bar"
[runtime]
   [[foo]]
       platform = nohosts-1
   [[bar]]
       platform = nohosts-2  # <----- or nohosts-1
```

With the two tasks on different platforms I get the expected result:
![trans-1](https://user-images.githubusercontent.com/2362137/228737685-ab5674d5-6954-4d27-800b-fc14575a000d.png)

With the two tasks on the same platform, one tries again (new job submit number) every second, indefinitely:

![trans-2](https://user-images.githubusercontent.com/2362137/228737820-e440e0bd-74d3-4e67-94b3-64be3c804775.png)

**Comment by oliver-sanders** [comment]
Probably best attempt after https://github.com/cylc/cylc-flow/pull/5395

**Comment by oliver-sanders** [comment]
@hjoliver I cannot reproduce this on upstream/8.1.x, can you confirm this.

If so, closed by #5395

**Comment by wxtim** [comment]
> @hjoliver I cannot reproduce this on upstream/8.1.x, can you confirm this.
> 
> If so, closed by #5395

I can confirm

**Comment by wxtim** [comment]
Closed by #5395
